06_puppy/nickolee/syncStore_test.go

```diff
@@ -0,0 +1,29 @@
+package puppystorer
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+/// Writing one test to give confidence that mapstore is retrieving puppies correctly
+func TestCreateAndRetrievePuppySyncStore(t *testing.T) {
```

Comment:
why not put this with the Suite?

Reply:
I had a look at this with @nicko-lee. He didn't have these tests to begin with. All tests were passing and had full coverage.
Unfortunately there was a bug in the way he was calculating the next ID (using the length of the current internal map). This cause the mapstore to overwrite old entries when you did `Create`, `Delete`, `Create`. I encouraged him to add this test and work on fixing the ID generation, and make sure it stays fixed with this test.

If you think this is overkill, I wouldn't bother with these tests. It just came out of the discussion that just using an interface test can result in 100% coverage, all tests pass, but still have fundamentally broken code.